### ISSUE TITLE
Please provide a cargo install option to not write .crates.toml

### ISSUE THREAD

**Issue by joshtriplett** [rust-lang/cargo#3316]
When installing a crate as part of a package build process (with `--root` pointing to a directory that will get packaged as `/usr`), I'd like to avoid writing the `.crates.toml` file (which would become `/usr/.crates.toml`).  Right now, I run `cargo install` and then remove that file afterward.  Please consider providing an option to not write `.crates.toml` at all.

**Comment by alexcrichton** [comment]
Sounds like a good option to me!

**Comment by cardoe** [comment]
Yes 👍 on this as well. For Gentoo I have to remove `.crates.toml` after the fact as well. (ref for me gentoo/gentoo#2918)

**Comment by JustAPerson** [comment]
@alexcrichton 
I was giving this a try since it looked relatively straight forward. Just a few quick questions.
I tried putting [this block](https://github.com/rust-lang/cargo/blob/db71d878fb4909575338f9f433b613ba94257d8b/src/cargo/ops/cargo_install.rs#L233-L239) in an if statement after adding a `--no-metadata` flag. However, it looks like the file is implicitly created elsewhere when trying to acquire a file lock. Perhaps just deleting the file at the very end of the `install()` function would work best?

While I'm at it, does this look fine for the updated `--help`?
```
...
    -q, --quiet               Less output printed to stdout
    --color WHEN              Coloring: auto, always, never
    --frozen                  Require Cargo.lock and cache are up to date
    --locked                  Require Cargo.lock is up to date
    --no-metadata             Do not write the .crates.toml file in the root
```
That's what I wrote before discovering the implicit creation. Should I specify it could delete an existing `.crates.toml` or should we check for the existence of the file before acquiring a lock and then only deleting the file if we created it?
